### ISSUE TITLE
READ/WRITE permissions not being set

### ISSUE THREAD

**Issue by anotherlab** [microsoft/WSA#258]
### Steps to reproduce

In WSA build 2301:
Write a .NET MAUI application that:
Sets the READ/WRITE permissions in the Platforms\Android\AndroidManifest.xml with the following syntax
```xml
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
And uses the CommunityToolkit.MAUI folder picker with the following syntax
```C#
CancellationTokenSource source = new();
CancellationToken cancellationToken = source.Token;

var folderPickerResult = await FolderPicker.PickAsync(cancellationToken);

if (folderPickerResult.IsSuccessful) {
  await Toast.Make($"Folder picked: Name - {folderPickerResult.Folder.Name}, Path - {folderPickerResult.Folder.Path}", ToastDuration.Long).Show(cancellationToken);
}
else {
  await Toast.Make($"Folder is not picked, {folderPickerResult.Exception.Message}").Show(cancellationToken);
}
```
I have taken the sample .NET MAUI project and replaced the sample code with a button that will invoke the filepicker.  This projects can be cloned from https://github.com/anotherlab/wsapermissions

### ✔️ Expected Behavior

Expected to get a "Allow XYZ to access photos and media on your device?" prompt, followed by a folder picker showing the same folder structure that would come up on an Android device or emulator.

This code works as expected for Android devices and emulators.

### ❌ Actual Behavior

Permissions prompt does not appear and call to FolderPicker.PickAsync comes back a FolderPickerResult.object with the IsSuccessful property set to false and the Exception.Message property set to "Storage permission is not granted."

If you set the permissions via adb with the syntax like the following:

    adb -s 127.0.0.1:58526 shell pm grant com.companyname.wsapermissions android.permission.READ_EXTERNAL_STORAGE

Then the folderpicker comes up in the expected folder.  You are still not prompted to give the app permissions until you actually pick a folder.

### Other Software

Using .NET MAUI 7 with Visual Studio 17.6.0 on Windows 11

### Please specify the version of Windows Subsystem for Android

2301.40000.7.0

**Comment by nieubank** [comment]
Thanks for reporting this issue. There is a change for apps targeting Android 13 related to granular media permissions that could be in play here.

Have you tried reproducing this on an Android 13 device, e.g. Pixel or otherwise.

**Comment by anotherlab** [comment]
I just tested on a Samsung Galaxy S22 running Android 13 and saw the same issue. I tried adding the granular media permissions [as described by Google](https://developer.android.com/about/versions/13/behavior-changes-13#granular-media-permissions), but that didn't change the problem. But it's not a WSA "thing", it's a CommunityToolkit thing.  I'll file this in that repo and this one can be closed as works as designed.
